api/processors/actions/group/discard.py

```diff
@@ -0,0 +1,62 @@
+from django.db import transaction
+
+from api.models import Group, Teacher
+from api.models.auxil.status_setter import StatusSetter
+from api.models.choices.log_event_type import StudentLogEventType, TeacherLogEventType
+from api.models.choices.status import StudentProjectStatus, TeacherProjectStatus
+from api.processors.actions.group import GroupActionProcessor
+from api.processors.auxil.log_event_creator import GroupLogEventCreator
+
+
+class GroupDiscardProcessor(GroupActionProcessor):
+    def __init__(self, group: Group, reason: str):
+        self.reason = reason
+        super().__init__(group)
+
+    @transaction.atomic
+    def process(self) -> None:
+        self._set_statuses()
+        self._create_log_events()
+        self._delete()
+
+    def _create_log_events(self) -> None:
+        GroupLogEventCreator.create(
+            group=self.group,
+            student_log_event_type=StudentLogEventType.TENTATIVE_GROUP_DISCARDED,
+            teacher_log_event_type=TeacherLogEventType.TENTATIVE_GROUP_DISCARDED,
+            comment=self.reason,
```

Comment:
Note that currently we can not add `group_from` due to `CASCAD` policy at `LogEvent`'s Models

Reply:
this should change in #209 (will be `SET_NULL` instead of `CASCADE`). I think that your PR might be merged earlier than #209, so maybe add a #TODO to add `group_from` once #209 is merged?